data_validation/query_builder/random_row_builder.py

```diff
@@ -43,6 +92,95 @@ def compile(
         table_name (String): The name of the table to query.
     """
         table = clients.get_ibis_table(data_client, schema_name, table_name)
-        query = table[self.primary_keys].limit(self.batch_size)
+        randomly_sorted_table = self.maybe_add_random_sort(data_client, table)
+        query = randomly_sorted_table.limit(self.batch_size)[self.primary_keys]
 
         return query
+
+    def maybe_add_random_sort(
+        self, data_client: ibis.client, table: ibis.Expr
+    ) -> ibis.Expr:
+        """ Return a randomly sorted query if it is supported for the client."""
+        if type(data_client) in RANDOM_SORT_SUPPORTS:
```

Comment:
Can we log a warning in case the data client does not support random sort?

Reply:
done